### PR TITLE
Payments Overview API caching

### PR DESCRIPTION
**Pull request by kidinov**
The PR adds API that allows us to cache response from the `overview-all` endpoint

Do not merge, as I want to see how that works in the app before

https://github.com/orgs/woocommerce/projects/185/views/1

### CODE REVIEW

**Comment by kidinov** [comment]
@AnirudhBhat fyi we will have to merge this before cut off otherwise we will break the build in the app

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/mappers/WooPaymentsDepositsOverviewMapper.kt [hunk 1]
```diff
@@ -0,0 +1,258 @@
+package org.wordpress.android.fluxc.persistence.mappers
```
Comment (AnirudhBhat) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/mappers/WooPaymentsDepositsOverviewMapper.kt:1]: 💡 This whole class could make use of some tests. Considering this has to go by this week, we can do this in a separate PR.

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/dao/WooPaymentsDepositsOverviewDao.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import org.wordpress.android.fluxc.persistence.entity.DepositType.LAST_PAID
+import org.wordpress.android.fluxc.persistence.entity.DepositType.NEXT_SCHEDULED
+import org.wordpress.android.fluxc.persistence.entity.WooPaymentsBalanceEntity
+import org.wordpress.android.fluxc.persistence.entity.WooPaymentsDepositEntity
+import org.wordpress.android.fluxc.persistence.entity.WooPaymentsDepositsOverviewEntity
+import org.wordpress.android.fluxc.persistence.entity.WooPaymentsManualDepositEntity
+
+@Dao
+interface WooPaymentsDepositsOverviewDao {
+    @Transaction
```
Comment (AnirudhBhat) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/dao/WooPaymentsDepositsOverviewDao.kt:21]: 👍

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/dao/WooPaymentsDepositsOverviewDao.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertDeposit(deposit: WooPaymentsDepositEntity)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertManualDeposit(manualDeposit: WooPaymentsManualDepositEntity)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertBalance(balance: WooPaymentsBalanceEntity)
+
+    @Transaction
```
Comment (AnirudhBhat) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/dao/WooPaymentsDepositsOverviewDao.kt:82]: 👍

**Review by AnirudhBhat — APPROVED** [review]
Thank you for working on this @kidinov. The changes LGTM! I've just left a minor comment that is non-blocking for the merge.

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/mappers/WooPaymentsDepositsOverviewMapper.kt (hunk 1 above)
Comment (kidinov) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/mappers/WooPaymentsDepositsOverviewMapper.kt:1]: I added a test  - 7cf443262f602eb083bd512e6edae973a0ddac0f
